coordinator/src/notification_service.rs

```diff
@@ -130,3 +144,205 @@ async fn send_notification<'a>(
     tracing::debug!("Sent: {:?}", response);
     Ok(())
 }
+
+/// Send notifications to users with positions that are about to expire or have
+/// just expired
+pub async fn send_expiry_notifications_if_applicable(
+    positions: &[(Position, FcmToken)],
+    notification_sender: &mpsc::Sender<Notification>,
+) {
+    let now = OffsetDateTime::now_utc();
+
+    for (position, fcm_token) in positions {
+        if position.expiry_timestamp <= now
+            && now < position.expiry_timestamp + time::Duration::hours(1)
+        {
+            if let Err(e) = notification_sender
+                .send(Notification::new(
+                    fcm_token.clone(),
+                    NotificationKind::PositionExpired,
+                ))
+                .await
+            {
+                tracing::error!("Failed to send PositionExpired notification: {:?}", e);
+            }
+        } else if position.expiry_timestamp > now
+            && position.expiry_timestamp <= now + time::Duration::hours(1)
+        {
+            if let Err(e) = notification_sender
+                .send(Notification::new(
+                    fcm_token.clone(),
+                    NotificationKind::PositionSoonToExpire,
+                ))
+                .await
+            {
+                tracing::error!("Failed to send PositionSoonToExpire notification: {:?}", e);
+            }
+        }
+    }
+}
+
+#[cfg(test)]
+pub mod tests {
+    use super::*;
+    use crate::logger::init_tracing_for_test;
+    use time::Duration;
+
+    fn soon_expiring_position() -> (Position, FcmToken) {
+        let mut position = Position::dummy();
+        position.creation_timestamp = OffsetDateTime::now_utc() - Duration::days(1);
```

Comment:
:thinking: Does the `creation_timestamp` matter at all?

Reply:
not really, I could remove them. I only set them as it's nicer to have more realistic data :) would you prefer if I removed this part?

Reply:
I prefer to remove everything that's not strictly necessary so that the test is simpler, but not a big deal.